src/python_framework_v02/troute/nhd_io.py

```diff
@@ -688,12 +710,42 @@ def get_usgs_from_time_slices_folder(
     last_obs_NaN = {"obs":NaN, "time":NaT}  # No valid recent observation
     ```
     """
-    # usgs_df = usgs_df.interpolate(method="linear", axis=1)
-    # usgs_df = usgs_df.interpolate(method="linear", axis=1, limit_direction="backward")
-    usgs_df.drop(usgs_df[usgs_df.iloc[:, 0] == -999999.000000].index, inplace=True)
 
-    return usgs_df
+    # ---- Laugh testing ------
+    # scren-out erroneous qc flags
+    usgs_qual_df = usgs_qual_df.mask(usgs_qual_df < 0, np.nan)
+    usgs_qual_df = usgs_qual_df.mask(usgs_qual_df > 1, np.nan)
+
+    # screen-out poor quality flow observations
+    # TODO: Bring qc_thresh parameter in from yaml as user input
+    qc_trehsh = 1
+    usgs_df = usgs_df.mask(usgs_qual_df < qc_trehsh, np.nan)
+
+    # screen-out erroneous flow observations
+    usgs_df = usgs_df.mask(usgs_df < 0, np.nan)
+
+    # ---- Interpolate USGS observations to time discretization of the simulation ----
+    usgs_df_T = usgs_df.transpose()
+    usgs_df_T.index = pd.to_datetime(usgs_df_T.index, format = "%Y-%m-%d_%H:%M:%S")
+
+    # TODO: Bring max_fill_lmin parameter in from yaml as user input
+    max_fill_1min = 59
+    """
+    Note: The max_fill is applied when the series is being considered at a 1 minute interval
+    14 minutes ensures no over-interpolation with 15-minute gage records, but creates
+    square-wave signals at gages reporting hourly.
+
+    TODO: Add reporting interval information to the gage preprocessing (timeslice generation)
+    """
+    usgs_df_T = (usgs_df_T.resample('min').
+                 interpolate(limit = max_fill_1min, limit_direction = 'both').
+                 resample('5min').
+                 asfreq().
+                 loc[date_time_start_center:,:])
```

Comment:
Temporally trim the dataframe, which should ensure that the first (interpolated) observations listed are on the same time as the center time of the (temporally) first TimeSlice file in the glob.